example/anchor.js

```diff
@@ -0,0 +1,172 @@
+//
+// The first, direct anchoring, places the content directly into a transaction. This method makes
+// the content public and immutable but direct anchors become expensive to post as content becomes
+// large. Direct anchors are also costly to post in bulk (i.e. anchoring multiple documents).
+//
+// It's advised to hash documents in order to keep anchoring price constant irrespective of document size.
+// It's advised to batch document hashes into a merkle tree to keep anchoring price constant
+// irrespective of document count.
+//
+// A potential downside to merkle batching is that PoE verification of each batched document will
+// require a merkle proof of inclusion. This is a tradeoff to consider when designing for your use
+// case. Merkleized anchors can batch infinite anchors into a single, fixed cost transaction, but
+// verification requires a proof.
+//
+// This example uses mrklt for merkle tree construction. A mrklt tree containing exactly one
+// document will have `root = hash(hash(document))` and the merkle proof for that document will be
+// an empty list `[]`. `verify_proof(hash(document), []) = root = hash(hash(document))`. In other
+// words, when batch size is 1, we can infer `proof = []`.
+//
+// It is possible simply post `hash(document)` as an anchor, but it's recommended to double-hash
+// the document instead. Since `hash(hash(document)) = compute_root(hash(document))` a double-hashed
+// anchor can be interpreted as the root of a merkle-tree with 1 leaf. Since the merkle tree has
+// only one leaf, the proof of inclusion for that leaf will be empty.
+
+/* eslint-disable import/no-extraneous-dependencies, import/newline-after-import */
+
+/* eslint-disable camelcase */
+import { compute_root, create_proof, verify_proof } from 'mrklt';
+/* eslint-disable camelcase */
+import assert from 'assert';
+import BLAKE2s from 'blake2s-js';
+import BLAKE2b from 'blake2b';
+import { randomAsU8a } from '@polkadot/util-crypto';
+import { u8aToHex } from '@polkadot/util';
+import { connect, keypair } from '../scripts/helpers';
+
+require('dotenv').config();
+const { FullNodeEndpoint, TestAccountURI } = process.env;
+const conn = connect(FullNodeEndpoint);
+
+async function main() {
+  // batched
+  const docHashes = [
+    utf8('{"example": "document"}'),
+    utf8('{"example": 2}'),
+    randomAsU8a(),
+    utf8('{"example": 4}'),
+  ].map(blake2s);
+  const proofs = await anchorBatched(docHashes);
+  assert(await checkBatched(docHashes[0], proofs[0]) !== null);
+  assert(await checkBatched(docHashes[0], proofs[1]) === null);
+
+  // single
+  const single = blake2s(randomAsU8a());
+  assert(await checkBatched(single, []) === null);
+  await anchorBatched([single]);
+  assert(await checkBatched(single, []) !== null);
+}
+
+// Post a value to the anchors module.
+async function anchor(hash) {
+  const nc = await conn;
+  await signExtrinsic(nc.tx.anchor.deploy(u8aToHex(hash)))
```

Comment:
Can you add an anchor module supporting read/write of anchors so caller don't need to use the low level api object like `api.tx.anchor` and `api.query.anchor.anchors` but can use `dock.anchor.write` and `dock.anchor.check` and the anchor can be passed as bytes or hex?

Reply:
This is possible and I can see it improving developer experience.

<details><summary>Side comment</summary>
I'm personally of the opinion that accepting arguments in multiple formats can be harmful.
</details>

The anchoring [rfc explicitly notes this as out-of-scope](https://github.com/docknetwork/planning/blob/master/rfc/0007-anchoring.md#non-goals). I think it should be a standalone task and not part of the milestone. Created ticket: https://github.com/docknetwork/sdk/issues/209